The git testsuite fails when git does not have user.{name,email} configuration set

`opam` fails to initialize in the test environment when git does not have `user.name` and `user.email` set, leading to the whole git testsuite failing.

This occured while packaging version 2.0.0 for Debian.

```
[31m>> init FAILED <<[0m
rm -rf /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO
mkdir -p /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO
cd /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO &&   git init && echo '*.sh text eol=lf'> .gitattributes && touch README &&  git add -A && git commit -m "Initial commit"
Initialized empty Git repository in /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/.git/

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address (got 'nicoo@neon.(none)')
make[5]: *** [Makefile:103: init] Error 128
[31minit FAILED[0m

[31m>> upload FAILED <<[0m
cd packages && tar czf P1-0.tar.gz P1-0
cd packages && tar czf P1-1.tar.gz P1-1
cd packages && tar czf P1-2.tar.gz P1-2
cd packages && tar czf P2.tar.gz P2
cd packages && tar czf P3.tar.gz P3
cd packages && tar czf P4.tar.gz P4
cd packages && tar czf P5.tar.gz P5
cp -r packages/ocaml /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages
mkdir -p /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P1.0
cp packages/P1-0.opam     /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P1.0/opam
echo 'src: "http://dev.null" checksum: "'`openssl md5 packages/P1-0.tar.gz |cut -d' ' -f2`'"' > /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P1.0/url
mkdir -p /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P1.1
cp packages/P1-1.opam     /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P1.1/opam
cp packages/P1-1/README   /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P1.1/descr
echo 'src: "http://dev.null" checksum: "'`openssl md5 packages/P1-1.tar.gz |cut -d' ' -f2`'"' > /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P1.1/url
mkdir -p /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P2.1
cp packages/P2/README     /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P2.1/descr
cp packages/P2.opam       /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P2.1/opam
echo 'src: "http://dev.null" checksum: "'`openssl md5 packages/P2.tar.gz |cut -d' ' -f2`'"' > /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P2.1/url
mkdir -p /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P3.1~weird-version.test
cp packages/P3.opam       /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P3.1~weird-version.test/opam
cp packages/P3/README     /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P3.1~weird-version.test/descr
echo 'src: "http://dev.null" checksum: "'`openssl md5 packages/P3.tar.gz |cut -d' ' -f2`'"' > /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P3.1~weird-version.test/url
mkdir -p /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P4.1
cp packages/P4-1.opam     /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P4.1/opam
cp packages/P4/README     /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P4.1/descr
echo 'src: "http://dev.null" checksum: "'`openssl md5 packages/P4.tar.gz |cut -d' ' -f2`'"' > /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P4.1/url
mkdir -p /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P5.1
cp packages/P5.opam       /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P5.1/opam
cp packages/P5/README     /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P5.1/descr
echo 'src: "http://dev.null" checksum: "'`openssl md5 packages/P5.tar.gz |cut -d' ' -f2`'"' > /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P5.1/url
cd /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/ocaml.system && git add * && git commit -m "Adding ocaml.system"

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address (got 'nicoo@neon.(none)')
make[5]: *** [Makefile:138: upload] Error 128
[31mupload FAILED[0m

[31m>> install-remove FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/tools/opam_check.exe -l install-remove-1 ocaml.system
00:00.001  SYSTEM                          error: File /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT/config does not exist or can't be read
Fatal error: exception File /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT/config does not exist or can't be read
Raised at file "src/core/opamSystem.ml", line 25, characters 4-30
Called from file "src/tools/opam_check.ml", line 46, characters 15-58
Called from file "src/tools/opam_check.ml", line 54, characters 18-30
make[5]: *** [Makefile:185: install-remove] Error 2
[31minstall-remove FAILED[0m

[31m>> list FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe list -A
00:00.002  GSTATE                          LOAD-GLOBAL-STATE @ /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT
[ERROR] Opam has not been initialised, please run `opam init'
Backtrace:
  Raised at file "src/core/opamStd.ml", line 851, characters 15-29
  Called from file "src/state/opamGlobalState.ml", line 52, characters 4-113
  Called from file "src/state/opamGlobalState.ml", line 157, characters 11-20
  Called from file "src/cmdliner_term.ml", line 27, characters 19-24
  Called from file "src/cmdliner.ml", line 106, characters 32-39
make[5]: *** [Makefile:182: list] Error 50
[31mlist FAILED[0m

[31m>> install-opt FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/tools/opam_check.exe -l install-opt-1 ocaml.system
00:00.000  SYSTEM                          error: File /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT/config does not exist or can't be read
Fatal error: exception File /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT/config does not exist or can't be read
Raised at file "src/core/opamSystem.ml", line 25, characters 4-30
Called from file "src/tools/opam_check.ml", line 46, characters 15-58
Called from file "src/tools/opam_check.ml", line 54, characters 18-30
make[5]: *** [Makefile:192: install-opt] Error 2
[31minstall-opt FAILED[0m

[31m>> list FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe list -A
00:00.002  GSTATE                          LOAD-GLOBAL-STATE @ /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT
[ERROR] Opam has not been initialised, please run `opam init'
Backtrace:
  Raised at file "src/core/opamStd.ml", line 851, characters 15-29
  Called from file "src/state/opamGlobalState.ml", line 52, characters 4-113
  Called from file "src/state/opamGlobalState.ml", line 157, characters 11-20
  Called from file "src/cmdliner_term.ml", line 27, characters 19-24
  Called from file "src/cmdliner.ml", line 106, characters 32-39
make[5]: *** [Makefile:182: list] Error 50
[31mlist FAILED[0m

[31m>> install FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/tools/opam_check.exe -l install-1 ocaml.system
00:00.000  SYSTEM                          error: File /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT/config does not exist or can't be read
Fatal error: exception File /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT/config does not exist or can't be read
Raised at file "src/core/opamSystem.ml", line 25, characters 4-30
Called from file "src/tools/opam_check.ml", line 46, characters 15-58
Called from file "src/tools/opam_check.ml", line 54, characters 18-30
make[5]: *** [Makefile:221: install] Error 2
[31minstall FAILED[0m

[31m>> list FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe list -A
00:00.003  GSTATE                          LOAD-GLOBAL-STATE @ /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT
[ERROR] Opam has not been initialised, please run `opam init'
Backtrace:
  Raised at file "src/core/opamStd.ml", line 851, characters 15-29
  Called from file "src/state/opamGlobalState.ml", line 52, characters 4-113
  Called from file "src/state/opamGlobalState.ml", line 157, characters 11-20
  Called from file "src/cmdliner_term.ml", line 27, characters 19-24
  Called from file "src/cmdliner.ml", line 106, characters 32-39
make[5]: *** [Makefile:182: list] Error 50
[31mlist FAILED[0m

[31m>> reinstall FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/tools/opam_check.exe -l reinstall-1 ocaml.system P1.1 P2.1 P3.1~weird-version.test P4.1
00:00.000  SYSTEM                          error: File /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT/config does not exist or can't be read
Fatal error: exception File /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT/config does not exist or can't be read
Raised at file "src/core/opamSystem.ml", line 25, characters 4-30
Called from file "src/tools/opam_check.ml", line 46, characters 15-58
Called from file "src/tools/opam_check.ml", line 54, characters 18-30
make[5]: *** [Makefile:232: reinstall] Error 2
[31mreinstall FAILED[0m

[31m>> list FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe list -A
00:00.004  GSTATE                          LOAD-GLOBAL-STATE @ /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT
[ERROR] Opam has not been initialised, please run `opam init'
Backtrace:
  Raised at file "src/core/opamStd.ml", line 851, characters 15-29
  Called from file "src/state/opamGlobalState.ml", line 52, characters 4-113
  Called from file "src/state/opamGlobalState.ml", line 157, characters 11-20
  Called from file "src/cmdliner_term.ml", line 27, characters 19-24
  Called from file "src/cmdliner.ml", line 106, characters 32-39
make[5]: *** [Makefile:182: list] Error 50
[31mlist FAILED[0m

[31m>> upload-new FAILED <<[0m
mkdir /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P4.2
cp packages/P4-2.opam   /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P4.2/opam
cp packages/P4/README   /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P4.2/descr
echo 'src: "http://dev.null" checksum: "'`openssl md5 packages/P4.tar.gz |cut -d' ' -f2`'"' > /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P4.2/url
mkdir /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P4.3
cp packages/P4-3.opam   /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P4.3/opam
cp packages/P4/README   /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P4.3/descr
echo 'src: "http://dev.null" checksum: "'`openssl md5 packages/P4.tar.gz |cut -d' ' -f2`'"' > /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO/packages/P4.3/url
echo "(* new line *)" >> /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.GIT/P1-1/p1.ml
/bin/sh: 1: cannot create /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.GIT/P1-1/p1.ml: Directory nonexistent
make[5]: *** [Makefile:245: upload-new] Error 2
[31mupload-new FAILED[0m

[31m>> list FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe list -A
00:00.004  GSTATE                          LOAD-GLOBAL-STATE @ /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT
[ERROR] Opam has not been initialised, please run `opam init'
Backtrace:
  Raised at file "src/core/opamStd.ml", line 851, characters 15-29
  Called from file "src/state/opamGlobalState.ml", line 52, characters 4-113
  Called from file "src/state/opamGlobalState.ml", line 157, characters 11-20
  Called from file "src/cmdliner_term.ml", line 27, characters 19-24
  Called from file "src/cmdliner.ml", line 106, characters 32-39
make[5]: *** [Makefile:182: list] Error 50
[31mlist FAILED[0m

[31m>> upgrade FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/tools/opam_check.exe -l upgrade-1 ocaml.system P1.1 P2.1 P3.1~weird-version.test P4.1
00:00.000  SYSTEM                          error: File /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT/config does not exist or can't be read
Fatal error: exception File /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT/config does not exist or can't be read
Raised at file "src/core/opamSystem.ml", line 25, characters 4-30
Called from file "src/tools/opam_check.ml", line 46, characters 15-58
Called from file "src/tools/opam_check.ml", line 54, characters 18-30
make[5]: *** [Makefile:260: upgrade] Error 2
[31mupgrade FAILED[0m

[31m>> list FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe list -A
00:00.002  GSTATE                          LOAD-GLOBAL-STATE @ /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT
[ERROR] Opam has not been initialised, please run `opam init'
Backtrace:
  Raised at file "src/core/opamStd.ml", line 851, characters 15-29
  Called from file "src/state/opamGlobalState.ml", line 52, characters 4-113
  Called from file "src/state/opamGlobalState.ml", line 157, characters 11-20
  Called from file "src/cmdliner_term.ml", line 27, characters 19-24
  Called from file "src/cmdliner.ml", line 106, characters 32-39
make[5]: *** [Makefile:182: list] Error 50
[31mlist FAILED[0m

[31m>> downgrade FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/tools/opam_check.exe -l downgrade-1 ocaml.system P1.1 P2.1 P3.1~weird-version.test P4.3
00:00.000  SYSTEM                          error: File /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT/config does not exist or can't be read
Fatal error: exception File /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT/config does not exist or can't be read
Raised at file "src/core/opamSystem.ml", line 25, characters 4-30
Called from file "src/tools/opam_check.ml", line 46, characters 15-58
Called from file "src/tools/opam_check.ml", line 54, characters 18-30
make[5]: *** [Makefile:272: downgrade] Error 2
[31mdowngrade FAILED[0m

[31m>> list FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe list -A
00:00.002  GSTATE                          LOAD-GLOBAL-STATE @ /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT
[ERROR] Opam has not been initialised, please run `opam init'
Backtrace:
  Raised at file "src/core/opamStd.ml", line 851, characters 15-29
  Called from file "src/state/opamGlobalState.ml", line 52, characters 4-113
  Called from file "src/state/opamGlobalState.ml", line 157, characters 11-20
  Called from file "src/cmdliner_term.ml", line 27, characters 19-24
  Called from file "src/cmdliner.ml", line 106, characters 32-39
make[5]: *** [Makefile:182: list] Error 50
[31mlist FAILED[0m

[31m>> switch-alias FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/tools/opam_check.exe -l switch-alias-1 ocaml.system P1.1 P2.1 P3.1~weird-version.test P4.2
00:00.000  SYSTEM                          error: File /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT/config does not exist or can't be read
Fatal error: exception File /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT/config does not exist or can't be read
Raised at file "src/core/opamSystem.ml", line 25, characters 4-30
Called from file "src/tools/opam_check.ml", line 46, characters 15-58
Called from file "src/tools/opam_check.ml", line 54, characters 18-30
make[5]: *** [Makefile:280: switch-alias] Error 2
[31mswitch-alias FAILED[0m

[31m>> list FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe list -A
00:00.002  GSTATE                          LOAD-GLOBAL-STATE @ /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT
[ERROR] Opam has not been initialised, please run `opam init'
Backtrace:
  Raised at file "src/core/opamStd.ml", line 851, characters 15-29
  Called from file "src/state/opamGlobalState.ml", line 52, characters 4-113
  Called from file "src/state/opamGlobalState.ml", line 157, characters 11-20
  Called from file "src/cmdliner_term.ml", line 27, characters 19-24
  Called from file "src/cmdliner.ml", line 106, characters 32-39
make[5]: *** [Makefile:182: list] Error 50
[31mlist FAILED[0m

[31m>> switch-env-packages FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/tools/opam_check.exe -l switch-env-packages-1 ocaml.system P1.1 P2.1
00:00.001  SYSTEM                          error: File /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT/config does not exist or can't be read
Fatal error: exception File /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT/config does not exist or can't be read
Raised at file "src/core/opamSystem.ml", line 25, characters 4-30
Called from file "src/tools/opam_check.ml", line 46, characters 15-58
Called from file "src/tools/opam_check.ml", line 54, characters 18-30
make[5]: *** [Makefile:297: switch-env-packages] Error 2
[31mswitch-env-packages FAILED[0m

[31m>> repo FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe repo add test2 /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.REPO -k git
00:00.002  GSTATE                          LOAD-GLOBAL-STATE @ /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT
[ERROR] Opam has not been initialised, please run `opam init'
Backtrace:
  Raised at file "src/core/opamStd.ml", line 851, characters 15-29
  Called from file "src/state/opamGlobalState.ml", line 52, characters 4-113
  Called from file "src/state/opamGlobalState.ml", line 157, characters 11-20
  Called from file "src/cmdliner_term.ml", line 27, characters 19-24
  Called from file "src/cmdliner.ml", line 27, characters 27-34
  Called from file "src/cmdliner.ml", line 106, characters 32-39
make[5]: *** [Makefile:303: repo] Error 50
[31mrepo FAILED[0m

[31m>> list FAILED <<[0m
PATH=/build/opam-lavcaK/opam-2.0.0/_build/install/default/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games OPAMDEBUG=2 OCAMLRUNPARAM=b OPAMKEEPBUILDDIR=1 OPAMROOT=/build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT OPAMSWITCH= OPAMNOBASEPACKAGES=1 OPAMYES=1 OPAM=/build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe /build/opam-lavcaK/opam-2.0.0/_build/default/src/client/opamMain.exe list -A
00:00.002  GSTATE                          LOAD-GLOBAL-STATE @ /build/opam-lavcaK/opam-2.0.0/_build/default/tests/tmp/OPAM.ROOT
[ERROR] Opam has not been initialised, please run `opam init'
Backtrace:
  Raised at file "src/core/opamStd.ml", line 851, characters 15-29
  Called from file "src/state/opamGlobalState.ml", line 52, characters 4-113
  Called from file "src/state/opamGlobalState.ml", line 157, characters 11-20
  Called from file "src/cmdliner_term.ml", line 27, characters 19-24
  Called from file "src/cmdliner.ml", line 106, characters 32-39
make[5]: *** [Makefile:182: list] Error 50
[31mlist FAILED[0m
```

--- a/.travis-ci.sh
+++ b/.travis-ci.sh
@@ -177,7 +177,7 @@ export OCAMLRUNPARAM=b
   else
     # Note: these tests require a "system" compiler and will use the one in $OPAMBSROOT
     OPAMEXTERNALSOLVER="$EXTERNAL_SOLVER" make tests ||
-      (tail -2000 _build/default/tests/fulltest-*.log; echo "-- TESTS FAILED --"; exit 1)
+      (tail -n 2000 _build/default/tests/fulltest-*.log; echo "-- TESTS FAILED --"; exit 1)
   fi
 )
 

--- a/tests/Makefile
+++ b/tests/Makefile
@@ -94,7 +94,8 @@ git:
 
 define GIT_INIT
   git init && echo '*.sh text eol=lf'> .gitattributes && $(if $1,touch $1 && )\
-	git add -A && git commit -m "Initial commit"
+	git add -A && git config --local user.name 'OPAM test environment' &&\
+	git config --local user.email noreply@ocaml.org && git commit -m "Initial commit"
 endef
 
 init:

